File: _.kt
sealed class Student(val indexNumber: String)

class GraduateStudent(indexNumber: String, val graduationYear: Int) : Student(indexNumber)
class UnderGraduateStudent(indexNumber: String, val currentGrade: Int) : Student(indexNumber)

fun main(args: Array<String>) {
    val student: Student = UnderGraduateStudent("123", 3)
    when (student) {
        is UnderGraduateStudent -> println("UnderGraduateStudent :: indexNumber: ${student.indexNumber} :: current grade : ${student.currentGrade}")
        is GraduateStudent -> println("GraduateStudent :: indexNumber: ${student.indexNumber} :: graduation year : $ {student.currentGrade}")
        else -> println("unknown")
    }
}

File: _.rs
enum Student {
    GraduateStudent(String,i64)
    UnderGraduateStudent(String,i64)
}

fn main() {
    let mut s = Student::UnderGraduateStudent("123".to_string(),3);
    match s {
        Student::GraduateStudent(name,year) => {
            println!("GraduateStudent :: indexNumber: {} ::  graduation year : {}",name,year);
        }
        Student::UnderGraduateStudent(name,id) => {
            println!("UnderGraduateStudent :: indexNumber: {} :: current grade : {}",name,id);
        }
    }
}